### ISSUE TITLE
Sending badRequest when unauthorized should have been sent

### ISSUE THREAD

**Issue by anasyb** [sane/sane-auth#9]
ember-simple-auth expects an unauthorized access (401) to be sent when an issue happens with the auth. At the moment sails is sending a badRequest (400).

_Why?_
- In response to receiving 401, simple-auth "invalidates the session and reloads the application" [ref](http://ember-simple-auth.com/ember-simple-auth-api-docs.html#SimpleAuth-ApplicationRouteMixin-authorizationFailed).

_Fix_
- update `AuthController.js` to call res.unauthorized instead of res.badRequest
- include an api/responses/unauthorized.js file

**Comment by mgenev** [comment]
good call, feel free to submit a PR or I'll adjust it when i get back to Seattle

**Comment by Globegitter** [comment]
@anasyb @mgenev @kriswill  What do you think. Should we include the unauthorized.js already in sane core or better here in the addon?

**Comment by kriswill** [comment]
I think it's a good to have it as the base sails project.  Someone might have some other rule that they use to determine authorization besides roles, like resources owned by a user, etc.

**Comment by anasyb** [comment]
It might be the case people would find a use for it other than that associated with authentication, but I don't believe that's the intent of it. SO has [a summary](http://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses/6937030#6937030) on that and [IEFT states clearly](http://tools.ietf.org/html/draft-ietf-httpbis-p7-auth-25#section-3.1) it should be basically only used when credentials are provided but there is a a problem with them... the server asks the user to retry. In response simple-auth would invalidate the session and transfer the user to login. This behaviour shouldn't IMHO happen in other scenarios like the one mentioned above: resources not owned by a user. For those the forbidden 403 exist which says: I know exactly who you are and you're not allowed, don't even try!

That being said I don't know what's the answer to your question @Globegitter... I would personally put it here in sane-auth.

**Comment by kriswill** [comment]
@anasyb If the ONLY use of 401 is specifically for authentication, then it does belong to the addon project.

**Comment by Globegitter** [comment]
Yep I think I will plan it in for the auth addon for now. Can still change in the future if necessary.
